# Structure et réseau cristallins
L'objectif ici est de caractériser la structure et réseau cristallins du cristal [CdBr2](https://materialsproject.org/materials/mp-568690/); on s'intéresse pour celà aux vecteurs de bases des réseaux direct et réciproque, le type de maille, le système cristallin et le groupe ponctuel. Le groupe d'espace ainsi que l'ensemble des symétries relatif à la structure sont décrits dans le notebook [symmetry](symmetry.ipynb)

Il faut en premier lieu obtenir la structure via [MaterialsProject](https://materialsproject.org) et la convertir également en fichier CIF:

In [1]:
from pymatgen.ext.matproj import MPRester
from pymatgen.io.cif import CifWriter
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
import math
# Permet d'utiliser l'interface du MP
with MPRester("2r9TvKUOgDsTx87F")as mpr:
    
    # Permet d'obtenir l'objet 'structure' sur le MP
    structure = mpr.get_structure_by_material_id("mp-568690")
    new_s=SpacegroupAnalyzer(structure).get_primitive_standard_structure()
    sym=SpacegroupAnalyzer(structure).get_refined_structure() #Permet d'obtenir la maille hexagonale
    # Crée un objet permettant de convertir la structure en fichier cif
    cwsym=CifWriter(structure,0.1) #Fichier avec maille hexagonale
    cwsym.write_file("CdBr2sym")
    cwprim=CifWriter(new_s)     #Fichier avec maille primitive
    cwprim.write_file("CdBr2prim")
    

Deux fichiers cif ont été écrits, un pour la maille primitive de la structure et un pour la maille conventionnelle comme nous le verrons dans le point suivant.

## Vecteurs de bases

La librairie [pymatgen](https://pymatgen.org/) permet d'obtenir les caractéristiques des vecteurs de bases (longueurs,angles et volume de la maille):


In [12]:
# Donne les longueurs, angles et volume des différentes maille

       #Maille primitive 
lat=new_s.lattice
l=lat.abc
ang=lat.angles
omega=lat.volume
       #Maille Conventionnelle
lat2=sym.lattice
L=lat2.abc
Ang=lat2.angles
Omega=lat2.volume

       #Réseau réciproque, maille primitive
latrec=lat.reciprocal_lattice
lr=latrec.abc
angr=latrec.angles
omegar=latrec.volume

print('Maille primitive:')
print('longueurs: la=',l[0],';','lb=',l[1],';', 'lc=',l[2])
print('angles: alpha=',ang[0],';','beta=',ang[1],';', 'gamma=',ang[2])
print('Volume de la maille:',omega)
print()
print('Maille conventionnelle:')
print('longueurs: lA=',L[0],';','lB=',L[1],';', 'lC=',L[2])
print('angles: alpha=',Ang[0],';','beta=',Ang[1],';', 'gamma=',Ang[2])
print('Volume de la maille:',Omega)
print()
print('Réseau réciproque, maille primitive uniquement:')
print('longueurs: la*=',lr[0],';','lb*=',lr[1],';', 'lc*=',lr[2])
print('angles: alpha=',angr[0],';','beta=',angr[1],';', 'gamma=',angr[2])
print('Volume de la maille:',omegar)


Maille primitive:
longueurs: la= 7.246402217449919 ; lb= 7.246402217449919 ; lc= 7.246402217449919
angles: alpha= 32.67990124662151 ; beta= 32.67990124662151 ; gamma= 32.67990124662154
Volume de la maille: 98.67117723279073

Maille conventionnelle:
longueurs: lA= 4.07734594722864 ; lB= 4.07734594722864 ; lC= 20.560127819481632
angles: alpha= 90.0 ; beta= 90.0 ; gamma= 120.00000000000001
Volume de la maille: 296.01353169837216

Réseau réciproque, maille primitive uniquement:
longueurs: la*= 1.8054440347937817 ; lb*= 1.8054440347937821 ; lc*= 1.805444034793782
angles: alpha= 117.19520382564485 ; beta= 117.19520382564485 ; gamma= 117.19520382564485
Volume de la maille: 2.5139075097602634


La maille conventionnelle (hexagonale) a été calculée en plus car elle est plus commode pour décrire les symétries des groupes ponctuels et d'espaces de la structure, elle sera donc utilisée par la suite. Il n'est par contre pas nécessaire de donner son réseau réciproque.


Les vecteurs primitifs ont donc comme caractéristiques:
* $l_a=l_b=l_c=7.246 Å$  
* $\alpha=\beta=\gamma=32.68°$
* $\Omega=98.671 Å^3$

Ceci correspond à une maille rhombohédrique.



Les vecteurs de la maille conventionnelle sont caractérisés par:
* $l_A=l_B=4.077Å;$  $l_C=20.560Å$  
* $\alpha=\beta=90°;\gamma=120°$ 
* $\Omega=296.014 Å^3$

et correspond donc à une maille hexagonale.

Enfin, le réseau réciproque standard (avec un facteur $2\pi$) présente les caractéristiques: 
* $l_{a*}=l_{b*}=l_{c*}=1.805 Å$ 
* $\alpha=\beta=\gamma=117.20°$.
* $\Omega=2.514Å^3 $

Les vecteurs correspondants (exprimés dans une base orthonormée cartésienne) peuvent également être obtenus:


In [21]:
 # Donne les vecteurs primitifs du réseau direct et réciproque
    #Maille primitive
a=lat.get_cartesian_coords([1,0,0])
b=lat.get_cartesian_coords([0,1,0])
c=lat.get_cartesian_coords([0,0,1])
    
    #Maille conventionnelle
A=lat2.get_cartesian_coords([1,0,0])
B=lat2.get_cartesian_coords([0,1,0])
C=lat2.get_cartesian_coords([0,0,1])
    
    #Réseau réciproque, maille primitive uniquement
ar=latrec.get_cartesian_coords([1,0,0])
br=latrec.get_cartesian_coords([0,1,0])
cr=latrec.get_cartesian_coords([0,0,1])
    
print('Maille primitive:')
print('a=',l[0],a/l[0])
print('b=',l[1],b/l[1])
print('c=',l[2],c/l[2])
print()
print('Maille conventionelle:')
print('A=',L[0],A/L[0])
print('B=',L[1],B/L[1])
print('C=',L[2],C/L[2])
print()
print('Réseau réciproque, maille primitive uniquement:')
print('a*=',lr[0],ar/lr[0])
print('b*=',lr[1],br/lr[1])
print('c*=',lr[2],cr/lr[2])


Maille primitive:
a= 7.246402217449919 [ 0.95960936 -0.28133588  0.        ]
b= 7.246402217449919 [0.95960936 0.28133588 0.        ]
c= 7.246402217449919 [0.877128   0.         0.48025667]

Maille conventionelle:
A= 4.07734594722864 [1. 0. 0.]
B= 4.07734594722864 [-0.5        0.8660254  0.       ]
C= 20.560127819481632 [0. 0. 1.]

Réseau réciproque, maille primitive uniquement:
a*= 1.8054440347937817 [ 0.25023551 -0.85352899 -0.45702347]
b*= 1.8054440347937821 [ 0.25023551  0.85352899 -0.45702347]
c*= 1.805444034793782 [0. 0. 1.]


Les vecteurs de bases sont calculés ci-dessus (l'unité étant l'Angström) et représentés par le vecteur normalisé multiplié par sa longueur. Les mailles des réseaux directs sont représentées ci-dessous avec leurs contenus:

| <img src="Images/mailleprim1.PNG" alt="drawing" width="300"/>  | <img src="Images/mailleconv.PNG" alt="drawing" width="300"/> |
|:---:|:---:|
| Maile primitive(trigonal) du CdBr2, les atomes de Cd sont représentés en mauve tandis que les atomes de Br sont en brun|Maille conventionnelle (hexagonale) utilisée notamment pour décrire le groupe d'espace de la structure cristalline  | 

Pour comprendre le lien existant entre la maille primitive et conventionnelle il suffit de réaliser le changement de base suivant exprimant les vecteurs de la maille conventionnelle selon les vecteurs primitifs:

* $A=a-b$
* $B=b-c$
* $C=a+b+c$

|<img src="Images/trighex.png" alt="drawing" width="400"/>|
|:---:|
|La maille primitive ( schéma de gauche en rouge) est représenté dans un polyhèdre à base hexagonale formé par la maille conventionnelle (vecteurs A B C)|

## Système cristallin,système réticulaire et groupe ponctuel

Le système cristallin est basé sur une classification selon les groupes ponctuels et aspects morphologique des cristaux; le système réticulaire est une classification selon le type de réseau et les relations entre les vecteurs primitifs tandis que le groupe ponctuel est le groupe (au sens mathémathique) associé aux opérations de symétries ponctuels permises par le réseau. 


In [32]:
# Donne le syteme cristallin, le type de maille et le groupe ponctuel
struct=SpacegroupAnalyzer(structure)
lattype=struct.get_lattice_type()
system=struct.get_crystal_system()
pointg=struct.get_point_group_symbol()
print('Système cristallin:',system)
print('Système rériculaire:',lattype)
print('Groupe d espace:',pointg)

Système cristallin: trigonal
Système rériculaire: rhombohedral
Groupe d espace: -3m


### Système cristallin
Le cristal possède un système **trigonal**,système caractérisé principalement par le fait qu'il possède un axe de rotation ou roto-inversion d'ordre 3. Il possède les 5 groupes ponctuels suivants (notation de Hermann-Mauguin): **3, 3̅, 32, 3m, et 3̅m**

### Système réticulaire
Ce sont les paramètres de la maille primitive qui déterminent le système réticulaire. Ces paramètres sont tels que les vecteurs primitifs ont comme caractéristiques $l_a=l_b=l_c$ et $\alpha=\beta=\gamma \neq 90°$ (voir partie "vecteurs de bases". Celà correspond bien par définition à un système réticulaire de type **rhombohédrique**

### Groupe ponctuel

La structure cristalline possède comme éléments générateurs de symétries ponctuels un **axe de roto-inversion d'ordre 3** (dans la direction vecteur C de la maille hexagonale) et un **plan miroir** perpendiculaire à l'axe. Ces deux générateurs engendrent le groupe ponctuel **3̅m** d'ordre 12.


|<img src="Images/grouppoint.PNG" alt="drawing" width="500"/>|
|:---:|
Représentation visuelle du groupe 3̅m, les deux éléments de symétries générateurs du groupe sont affichés et les points peuvent représenter un noeud du réseau ou même un atome mais est surtout une représentation plus visuelle des 12 éléments du groupe